### PR TITLE
Support running Alluxio in intellij by import run configuration

### PR DESCRIPTION
**Pull request by runzhiwang**
**Is your feature request related to a problem? Please describe.**

For newbies of Alluxio, it will help them to start alluxio in the intellij easily by import the run configuration.

Leverage this, it can make it possible to debug alluxio in the intellij also.

**Describe the solution you'd like**

Write a script which can import the alluxio run configuration to the alluxio project .idea folder.

**Describe alternatives you've considered**

Put the script and run configurations into dev/intellij folder

**Urgency**

Middle

**Additional context**

Everybody will like this.

### CODE REVIEW

**Comment by maobaolong** [comment]
@runzhiwang Please add Fix #12737 at the start of the description.

**Review by Xenorith — CHANGES_REQUESTED** [review]
for other people to discover this, can you add a section to the documentation with instructions on how to utilize your changes? i think a good place would be here:
https://docs.alluxio.io/os/user/stable/en/contributor/Contributor-Tools.html#intellij-idea
which corresponds to the file at `docs/en/contributor/Contributor-Tools.md`

**Comment by runzhiwang** [comment]
@Xenorith Thanks the suggestion, will fix it.

**Comment by apc999** [comment]
@runzhiwang  thanks for your contribution!
do you get a chance to update this PR?

**Comment by runzhiwang** [comment]
@apc999 Thanks for reminding, and sorry for the delay, I will update this PR in one week.

**Review by Trevor-zhang — COMMENTED** [review]
good job .

File: dev/intellij/install-runconfig.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+# (the "License"). You may not use this work except in compliance with the License, which is
+# available at www.apache.org/licenses/LICENSE-2.0
+#
+# This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+# either express or implied, as more fully set forth in the License.
+#
+# See the NOTICE file distributed with this work for information regarding copyright ownership.
+#
+
+SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
```
Comment (gpang) [inline dev/intellij/install-runconfig.sh:13]: Can you add some comments to the top of this file which describes what this script does and how/where to run it?

**Review by gpang — CHANGES_REQUESTED** [review]
@runzhiwang Thanks! I left a comment.

File: dev/intellij/install-runconfig.sh (hunk 1 above)
Comment (runzhiwang) [reply dev/intellij/install-runconfig.sh:13]: @gpang Thanks for review, I will do it.

**Comment by apc999** [comment]
@runzhiwang will you plan to update this PR and address the comments?

**Comment by runzhiwang** [comment]
Fix #12721

**Comment by apc999** [comment]
there are some minor format issues left as commented by reviewers, can you address them @runzhiwang ?

**Comment by runzhiwang** [comment]
@apc999 Sorry for the delay, I have added the doc and fix the style.

File: docs/en/contributor/Contributor-Tools.md [hunk 2]
```diff
@@ -51,6 +51,29 @@ And then check the box next to "developer" in the window pane.
 > Note: Alluxio 2.2 moved generated gRPC proto source files into `alluxio/core/transport/target/generated-sources/protobuf/`.
 You will need to mark the directory as "Generated Sources Root" for IntelliJ to resolve the source files.
 
+#### Run in IntelliJ IDEA
```
Comment (gpang) [inline docs/en/contributor/Contributor-Tools.md:54]:
```suggestion
#### Run Alluxio processes within IntelliJ IDEA
```

File: docs/en/contributor/Contributor-Tools.md [hunk 3]
```diff
@@ -51,6 +51,29 @@ And then check the box next to "developer" in the window pane.
...
+1. Run `dev/intellij/install-runconfig.sh`
+2. Restart IntelliJ IDEA
+3. Edit conf/alluxio-site.properties
```
Comment (gpang) [inline docs/en/contributor/Contributor-Tools.md:57]:
```suggestion
3. Edit `conf/alluxio-site.properties`
```

File: docs/en/contributor/Contributor-Tools.md [hunk 4]
```diff
@@ -51,6 +51,29 @@ And then check the box next to "developer" in the window pane.
...
+```console
+  alluxio.home={alluxio.home}
+  alluxio.master.hostname=localhost
+  alluxio.master.journal.type=UFS
+```
+4. Edit conf/log4j.properties to print log in console
```
Comment (gpang) [inline docs/en/contributor/Contributor-Tools.md:63]:
```suggestion
4. Edit `conf/log4j.properties` to print log in console
```

File: docs/en/contributor/Contributor-Tools.md [hunk 5]
```diff
@@ -51,6 +51,29 @@ And then check the box next to "developer" in the window pane.
...
+```console
+  log4j.rootLogger=INFO, ${alluxio.logger.type}, ${alluxio.remote.logger.type}, stdout
+  log4j.threshold=ALL
+  log4j.appender.stdout=org.apache.log4j.ConsoleAppender
+  log4j.appender.stdout.layout=org.apache.log4j.PatternLayout
+  log4j.appender.stdout.layout.ConversionPattern=%d{ISO8601} %-5p %c{2} (%F:%M(%L)) - %m%n
+```
+5. bin/alluxio formatMaster
```
Comment (gpang) [inline docs/en/contributor/Contributor-Tools.md:71]:
```suggestion
5. Format the Alluxio master by running `bin/alluxio formatMaster`
```

File: docs/en/contributor/Contributor-Tools.md [hunk 6]
```diff
@@ -51,6 +51,29 @@ And then check the box next to "developer" in the window pane.
...
+6. `Run > Run > AlluxioMaster`
+7. bin/alluxio-mount.sh SudoMount && bin/alluxio formatWorker
```
Comment (gpang) [inline docs/en/contributor/Contributor-Tools.md:73]:
```suggestion
7. Format the Alluxio Worker with `bin/alluxio-mount.sh SudoMount && bin/alluxio formatWorker`
```

File: docs/en/contributor/Contributor-Tools.md [hunk 7]
```diff
@@ -51,6 +51,29 @@ And then check the box next to "developer" in the window pane.
...
+4. Edit conf/log4j.properties to print log in console
+```console
+  log4j.rootLogger=INFO, ${alluxio.logger.type}, ${alluxio.remote.logger.type}, stdout
+  log4j.threshold=ALL
+  log4j.appender.stdout=org.apache.log4j.ConsoleAppender
+  log4j.appender.stdout.layout=org.apache.log4j.PatternLayout
+  log4j.appender.stdout.layout.ConversionPattern=%d{ISO8601} %-5p %c{2} (%F:%M(%L)) - %m%n
+```
+5. bin/alluxio formatMaster
+6. `Run > Run > AlluxioMaster`
```
Comment (gpang) [inline docs/en/contributor/Contributor-Tools.md:72]: What exactly is this? Are these buttons in Intellij? If so, this should be something like 
`In Intellij, start Alluxio master process by selecting Run > Run > AlluxioMaster`

File: docs/en/contributor/Contributor-Tools.md [hunk 8]
```diff
@@ -51,6 +51,29 @@ And then check the box next to "developer" in the window pane.
...
+7. bin/alluxio-mount.sh SudoMount && bin/alluxio formatWorker
+8. `Run > Run > AlluxioWoker`
```
Comment (gpang) [inline docs/en/contributor/Contributor-Tools.md:74]: Can this be something like 
`In Intellij, start Alluxio worker process by selecting Run > Run > AlluxioWorker`

**Review by gpang — CHANGES_REQUESTED** [review]
@runzhiwang Thanks for the docs! I left some comments.

File: docs/en/contributor/Contributor-Tools.md (hunk 7 above)
Comment (runzhiwang) [reply docs/en/contributor/Contributor-Tools.md:72]: Yes, these are buttons in intellij

**Comment by runzhiwang** [comment]
@gpang Thanks the suggestions, I have updated the patch.

**Review by Xenorith — APPROVED** [review]
looks good to me, thanks for both the intellij setup and docs update

**Review by gpang — APPROVED** [review]
LGTM

**Comment by apc999** [comment]
alluxio-bot, merge this please

**Pull request merged**